absorber_ds/absorber.py

```diff
@@ -0,0 +1,150 @@
+    * name of PLCs for changing state of absorber (inserted or extracted)
+    * attribute for holding value of absorber state (inserted or extracted)
+    * methods to Insert or Extract the absorber itself
+    """
+    # device initialization
+
+    def safe_init_device(self):
+        """
+        This is safe_init_device method overriden from Facade base class. It's
+        used to initialize facade device safely
+        """
+        super(Absorber, self).safe_init_device()
+        self.set_state(DevState.ON)
+        self.set_status("Device is running")
+
+    # proxy attributes
+
+    waterFlowAlarm_Achromat = proxy_attribute(
+        dtype=bool,
+        access=AttrWriteType.READ,
+        property_name="PlcAttrName_AchromatFSW",
+        description="PLC device and attribute name for alarm from water flow "
+                    "in the achromat")
+
+    PlcAttrName_Extract = proxy_attribute(
+        dtype=bool,
+        access=AttrWriteType.READ_WRITE,
+        property_name="PlcAttrName_ExtractAttribute",
+        description="PLC device and attribute name for extracting absorber")
+
+    PlcAttrName_Insert = proxy_attribute(
+        dtype=bool,
+        access=AttrWriteType.READ_WRITE,
+        property_name="PlcAttrName_InsertAttribute",
+        description="PLC device and attribute name for inserting absorber")
+
+    waterFlowAlarm_1 = proxy_attribute(
+        dtype=bool,
+        access=AttrWriteType.READ,
+        property_name="PlcAttrName_FSW_1",
+        description="PLC device and attribute name for flow alarm")
+
+    waterFlowAlarm_2 = proxy_attribute(
+        dtype=bool,
+        access=AttrWriteType.READ,
+        property_name="PlcAttrName_FSW_2",
+        description="PLC device and attribute name for flow alarm")
+
+    waterFlowAlarm_3 = proxy_attribute(
+        dtype=bool,
+        access=AttrWriteType.READ,
+        property_name="PlcAttrName_FSW_3",
+        description="PLC device and attribute name for flow alarm")
+
+    PlcAttrName_StateExtracted = proxy_attribute(
+        dtype=bool,
+        access=AttrWriteType.READ,
+        property_name="PlcAttrName_StateExtractedAttribute",
+        description="PLC device and attribute name for extracted state")
+
+    PlcAttrName_StateInserted = proxy_attribute(
+        dtype=bool,
+        access=AttrWriteType.READ,
+        property_name="PlcAttrName_StateInsertedAttribute",
+        description="PLC device and attribute name for inserted state")
+
+    # logical attributes
+
+    @logical_attribute(
+        dtype=bool,
+        bind=['PlcAttrName_StateExtracted', 'PlcAttrName_StateInserted'],
+        description="True: the absorber is inserted. False: the absorber "
+                    "is extracted. Based on PlcAttrName_StateExtracted and "
+                    "PlcAttrName_StateInserted")
+    def InExStatus(self, ins, exs):
+        return ins and not exs
+
+    # commands
+
+    @command
+    def Extract(self):
```

Comment:
The `Extract` and `Insert` commands should be written as `proxy_commands` with `write_argument` field set to `True`. More on that in the second example [here](http://tango-facadedevice.readthedocs.io/en/latest/tutorial.html#proxy-commands).